### PR TITLE
New libvirt policy

### PR DESCRIPTION
**Pull request by 5umm3r15**
New SELinux policy for libvirt drivers:
Hypervisor drivers:

    virtqemud (QEMU/KVM)
    virtlxcd (LXC)
    virtvboxd (VirtualBox)

Secondary drivers:

    virtstoraged (host storage mgmt)
    virtnetworkd (virtual network mgmt)
    virtinterface (network interface mgmt)
    virtnodedevd (physical device mgmt)
    virtsecretd (security credential mgmt)
    virtnwfilterd (ip[6]tables/ebtables mgmt)
    virtproxyd (proxy daemon)

SELinux policy for virtvxz and virtxend has not been created yet.

Services are separated from the original virt files that are not libvirt related and virt_supplementary policy module is created.

Policy for libvirt drivers is designed, including modification of old virt interfaces and also creating new interfaces.

The policy is based on this PR: https://github.com/fedora-selinux/selinux-policy/pull/492 with updated commits

### CODE REVIEW

File: policy/modules/contrib/virt_supplementary.te [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+## Allow qemu-ga to run unconfined scripts
+## </p>
+## </desc>
+gen_tunable(virt_qemu_ga_run_unconfined, false)
+
+gen_require(`
+    class passwd passwd;
+')
+
+type qemu_exec_t;
```
Comment (zpytela) [inline policy/modules/contrib/virt_supplementary.te]: This type seems to be defined in virt.te as well.

File: policy/modules/kernel/devices.if [hunk 2]
```diff
@@ -5389,6 +5389,24 @@ interface(`dev_dontaudit_write_urand',`
 	dontaudit $1 urandom_device_t:chr_file write;
 ')
 
+########################################
+## <summary>
+##      Delete files in the dev/urandom directory.
```
Comment (zpytela) [inline policy/modules/kernel/devices.if]: Note /dev/urandom is a file

File: policy/modules/kernel/devices.if [hunk 3]
```diff
@@ -5389,6 +5389,24 @@ interface(`dev_dontaudit_write_urand',`
...
+##      Domain allowed access.
+##      </summary>
+## </param>
+#
+interface(`dev_delete_urand',`
+        gen_require(`
+                type urandom_device_t;
+        ')
+
+        allow $1 urandom_device_t:chr_file unlink;
```
Comment (zpytela) [inline policy/modules/kernel/devices.if:5407]: More permissions were not needed, like del_entry_dir_perms for the parent directory type?

File: policy/modules/contrib/virt_supplementary.te (hunk 1 above)
Comment (5umm3r15) [reply policy/modules/contrib/virt_supplementary.te]: fixed, thank you

File: policy/modules/kernel/devices.if (hunk 3 above)
Comment (5umm3r15) [reply policy/modules/kernel/devices.if:5407]: no

File: policy/modules/kernel/devices.if (hunk 2 above)
Comment (5umm3r15) [reply policy/modules/kernel/devices.if]: fixed, thank you

**Comment by zpytela** [comment]
LGTM now, thank you for the effort

**Pull request merged**